File: ts/management-api-v2/cm_api_v2_put_item.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
  environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
  apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

// Updates an existing content item
const response = await client.updateContentItem()
  .byItemId('f4b3fc05-e988-4dae-9ac1-a94aba566474')
  // .byItemCodename('my_article')
  // .byItemExternalId('59713')
  .withData(
    {
      name: 'On Roasts',
      codename: 'my_article_my_article'
    }
  )
  .toPromise();

// Creates a new content item
const response = await client.upsertContentItem()
  .byItemExternalId('59713')
  .withData(
    {
      name: 'On Roasts',
      codename: 'my_article_my_article',
      // 'type' is only required when creating a new content item
      type: 'article'
    }
  )
  .toPromise();

File: net/management-api-v2/cm_api_v2_put_item.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var identifier = Reference.ByExternalId("59713");
// var identifier = Reference.ById(Guid.Parse("f4b3fc05-e988-4dae-9ac1-a94aba566474"));
// var identifier = Reference.ByCodename("my_article");
var upsertedItemResponse = await client.UpsertContentItemAsync(identifier, new ContentItemUpsertModel
{
    Name = "On Roasts",
    Codename = "my_article_my_article",
    Collection = Reference.ByCodename("default"),
    // 'Type' is only required when creating a new content item
    Type = Reference.ByCodename("article"),
});
